### DISCUSSION TITLE
Is there any API to check how many tasks in the taskList?

### DISCUSSION THREAD

**Discussion by maeer** [uber/cadence#4485] [Q&A]
Hi~
I want to know is there any API to check how many tasks in the taskList(with parameter domain and taskListName)?
I didn‘t find yet.

**Comment by longquanzheng** [comment]
There is no such api. (Though internally the PollForDecisionTask api can return a backlog counter hint but it’s only for sticky task. The counter is not persisted to db). 

Similar discussion https://github.com/uber/cadence/discussions/4460

**Reply by maeer** [reply]
ok, I have another question.
We check the tasks number from cassandra directly. But there is a problem that sometimes the number is wrong. Generally occurs when the number of queues is large, and then  we clear the queue by worker poll. Until the worker can not poll the ActivityTask, that's mean the number of the ActivityTask is 0, but we get the  number from cassandra is greater than 0. Is it because the cadence update is slow?

**Reply by longquanzheng** [reply]
Can you elaborate how you did it and what you got? 

If you use task_id field, that’s increasing globally across all tasklists. And it’s not guaranteed to increase by 1 every time

**Comment by longquanzheng** [comment]
What is your use case of getting the number? There is no such number maintained internally. And adding this support will be quit complicated— if especially for multiple partitions tasklist

**Reply by maeer** [reply]
I can accept a certain error in the number of taskList queues.
But I want to know if there is eventual consistency in the number of taskList queues.
That is to say, can the number of taskList queues reach a relatively correct value within a visible time range?
example: If there is 100 tasks in taskList, then 10 workers work on it, when there is 0 tasks, Can I get the value 0 from the cassandra?

**Reply by meiliang86** [reply]
In a steady system, you will continue to insert and consume tasks into/from the taskList. So at best this number will be an estimate.

**Reply by zedongh** [reply]
> In a steady system, you will continue to insert and consume tasks into/from the taskList. So at best this number will be an estimate.



> What is your use case of getting the number? There is no such number maintained internally. And adding this support will be quit complicated— if especially for multiple partitions tasklist


The number could be a worker resouces estimate metric for something like resource autoscale ability

**Reply by meiliang86** [reply]
Yes this number is great for making auto scaling decisions.

**Comment by meiliang86** [comment]
We don't have this capability for now, but it's on our roadmap.
The current thought on this is that the number will be approximate instead of accurate. But we will see what we can come up with in the end.
